modules/doc/content/framework_development/authorship_guidelines.md

```diff
@@ -0,0 +1,74 @@
+# MOOSE papers authorship guidelines
+
+MOOSE is a collaborative effort and not only do we always welcome contributions, but also want to
+credit them appropriately and in a sustainable manner. We do this by including our contributors
+in the relevant journal article(s) that should be used by all, both within the community and externally,
+to cite the MOOSE framework and its modules. Please see the [citing.md] page for the current versions
+of these articles. These articles are published and updated on a regular basis.
+
+## MOOSE framework paper authorship guidelines
+
+When citing MOOSE, whether when using it, using a derived application, or just referring to
+the MOOSE project, the top-level MOOSE paper is the article that should be cited. It is featured in the [Citing MOOSE](citing.md#MOOSE) page.
+This paper is regularly replaced with either a new paper or an "update" paper. "Update" papers
+refer to a summary of changes related to a periodic update of the software from the timeframe of the previous "update" paper, and can have a different authorship list as the original
+paper.
+
+In order to best acknowledge the current contributors to the framework, we strive to keep the authorship
+of the MOOSE framework paper representative. To do that, we publish a summary of the new developments in the framework
+and use the list of the principal contributors since the previous paper as the list of authors for the new paper.
+
+The contributors are listed in two categories.
+
+- Semi-permanent contributors are included in every new MOOSE framework paper by default. These contributors
+  have written a very large part of the MOOSE framework and must be credited, at least until their contributions
+  are overtaken by others.
+
+The current criteria for semi-permanent authorship are listed below:
+
+- Top 10 contributor as measured by lines of code in the `framework/` and other utility folders or
+- Top 10 contributor as measured by number of commits in the `framework/` folder
+
+
+!alert note
+While it seems poor practice to recognize contribution by lines of code or commit numbers, this criteria did not
+exist when these contributions were made, so the metric did not influence the coding style of these authors.
+These criteria may be reformulated in the future in terms of contributed features and impact.
```

Comment:
I don't think this kind of note is necessary. If you want a note here, I might say that pull request reviewers will ask that duplicate/extraneous code be removed or refactored and that meaningless commits be squashed.

Reply:
Or if you want to keep some of the content here that's ok, but I would consider mixing in my suggestions, and noting that we have always valued clean code and clean commit history (even if those values haven't 100% been honored all the time)

Reply:
re-worded. Maybe not everyone has to know that historically we did not use LOC / commits for authorship (though I suspect we did lol)